Dockerfile

```diff
@@ -35,7 +35,9 @@ RUN apt-get update -qq && \
         qemu-system-ppc \
         qemu-system-x86 \
         u-boot-tools \
-        xz-utils
+        xz-utils \
+     && rm -rf /var/lib/apt/list/*
```

Comment:
Building the Docker image with and without this commit doesn't actually change the size.

```
$ docker build --no-cache -t clangbuiltlinux:$(git show -s --format=%h) --build-arg LLVM_VERSION=8 .
```

First is with this commit and second is current master:

```
$ docker image list
REPOSITORY               TAG                 IMAGE ID            CREATED              SIZE
clangbuiltlinux          1c33dda             0a97df0db6da        About a minute ago   720MB
clangbuiltlinux          a114c0f             401dac588720        5 minutes ago        720MB
```

Did you do any other size tests?

Style nit: Place the `&&` before the `\` on the previous line.

Reply:
Please no branch compare, I'm and would repulse this thought strongly currently. Check the [dockerfile doc](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/), `/var/lib/apt/lists/` can be huge, this should always be there in the Dockerfile after an update/upgrade runs. Another discussion point is `dpkg -l`, instead of discussing on the ticket, I wanted the discussion on the pr. A different script from Dockerfile or in the Dockerfile? Do you know if this would affect the size?

Reply:
Please no branch compare, I'm and would repulse this thought strongly currently. Check the [dockerfile doc](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/), `/var/lib/apt/lists/` can be huge, this should always be there in the Dockerfile after an update/upgrade runs. Other discussion point is `dpkg -l`, instead of discussing on the ticket, I wanted the discussion on the pr. A different script from Dockerfile or in the Dockerfile? Do you know if this would affect the size?

Reply:
> Please no branch compare, I'm and would repulse this thought strongly currently.

How else would you suggest measuring the "slimming" that the commit message mentions?

> Check the [dockerfile doc](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/), `/var/lib/apt/lists/` can be huge, this should always be there in the Dockerfile after an update/upgrade runs.

That's fine, I feel like that should be in the commit message :) I'm fine with this change since it's a part of best practices.

> Other discussion point is `dpkg -l`, instead of discussing on the ticket, I wanted the discussion on the pr. A different script from Dockerfile or in the Dockerfile? Do you know if this would affect the size?

I don't think there is any point to printing out that information since we don't have any unneeded dependencies in our image after 6ed21a9aaadb98e1979ec634e9e51f6690cdd7b9.

Reply:
> > Please no branch compare, I'm and would repulse this thought strongly currently.

> How else would you suggest measuring the "slimming" that the commit message mentions?

Great point, I should have added this in the pr description. Currently we are discussing.

> > Other discussion point is `dpkg -l`, instead of discussing on the ticket, I wanted the discussion on the pr. A different script from Dockerfile or in the Dockerfile? Do you know if this would affect the size?

> I don't think there is any point to printing out that information since we don't have any unneeded dependencies in our image after 6ed21a9aaadb98e1979ec634e9e51f6690cdd7b9.

I read the commit, a dockerfile w/o `--no-install-recommends` 😄. Yea this does decrease the size. I wanted the discussion, `dpkg -l` might decrease the size not this largely but surely will. I describe a possible approach in the next/above comment.